java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthVector.java

```diff
@@ -445,7 +446,7 @@ private long computeAndCheckOffsetsBufferSize(int valueCount) {
      * an additional slot in offset buffer.
      */
     final long size = computeCombinedBufferSize(valueCount + 1, OFFSET_WIDTH);
-    if (size > MAX_ALLOCATION_SIZE) {
+    if (size > MAX_BUFFER_SIZE) {
       throw new OversizedAllocationException("Memory required for vector capacity " +
           valueCount +
           " is (" + size + "), which is more than max allowed (" + MAX_ALLOCATION_SIZE + ")");
```

Comment:
```suggestion
          " is (" + size + "), which is more than max allowed (" + MAX_BUFFER_SIZE + ")");
```

Reply:
I wonder whether the exception messages should point users to LargeVar*Vectors when exceeding buffer capacity. Based on my experiences with RDBMS, I expected LargeVarCharVector to be suitable for storing large values, and missed that it is needed for many small values as well. I'm not sure how well-understood this is, and perhaps users would benefit from being pointed in an appropriate direction.

Reply:
Agree. I think the error message should give more information about how to solve the problem. Updated the message to point to the LargeVar*Vectors.